### DISCUSSION TITLE
Scripting with OperationMask

### DISCUSSION THREAD

**Discussion by nmilnercollabora** [sn4k3/UVtools#510] [Scripts]
Hello,

Please could you let me know how I can use OperationMask inside a script in particular specify the mask .png file? 

```
	    var operation = new OperationMask(SlicerFile) 
	    {
	       //how to specify mask file?
	       
	    };            
            operation.SelectCurrentLayer(layerIndex);
	    operation.Execute(Progress);
```
Thanks,

Nick

**Comment by sn4k3** [comment]
You need to see the source file [OperationMask.cs](https://github.com/sn4k3/UVtools/blob/master/UVtools.Core/Operations/OperationMask.cs)

However i don't have any method to load the mask from file directly (I may add/change that in next patch)
You need to strict give the `Mat` which can be obtained by:

```C#
var operation = new OperationMask(SlicerFile) 
	    {
	       Mask = CvInvoke.Imread("path_to_image.png", ImreadModes.Grayscale);
	    };    
```

Note that Mask must be same size as your ROI or slicer resolution

Here the sample from UI:

https://github.com/sn4k3/UVtools/blob/54d0cc1776282cf951861d71c7b35caed15870e2/UVtools.WPF/Controls/Tools/ToolMaskControl.axaml.cs#L107

**Comment by nmilnercollabora** [comment]
Thank you @sn4k3 . I have to say I am very impressed with UVTools. I found out about it [via](https://blog.honzamrazek.cz/2022/02/a-step-by-step-guide-for-the-perfect-bed-adhesion-and-removing-elephant-foot-on-a-resin-3d-printer/) and it's the best post slicing tool I've ever used.

**Reply by sn4k3** [reply]
I'm biased, but UVtools is such a awesome art piece of software with ton of freedom and capabilities, really unvaluable.
